File: day1/solution.js
const fs = require("fs");

let data = fs.readFileSync("input.txt", "utf8");
data = data.split("\n");

// Question 1: Elf carrying most calories
const elfCals = [];

let currentCals = 0;
for (const cal of data) {
  if (cal) {
    currentCals += Number(cal);
  } else {
    elfCals.push(currentCals);
    currentCals = 0;
  }
}

console.log("Question 1: ", Math.max(...elfCals));

//  Question 2: Top 3 elf calories combined

console.log(
  "Question 2: ",
  elfCals
    .sort((a, b) => b - a)
    .slice(0, 3)
    .reduce((prev, curr) => prev + curr, 0)
);

File: day1/solution.py
with open('input.txt', 'r') as f:
    lines = f.readlines()

lines = [line.strip() for line in lines]

# Question 1: Elf carrying most calories
current_cals = 0
elf_cals = []

for cal in lines:
    if cal:
        current_cals += int(cal)
    else:
        elf_cals.append(current_cals)
        current_cals = 0
        
print("Question 1: ", max(elf_cals))

# Question 2: Top 3 elf calories combined

print("Question 2: ", sum(sorted(elf_cals)[-3:]))